### PR TITLE
Docker-Trestle integration

### LINKED ISSUE #94: Switch OSCAL from compliance-template to docker-trestle
Utilize docker-trestle instead of compliance-template for OSCAL control text.

Look into packaging all controls as components instead of inserting content in markdown files.

### PR DESCRIPTION
**Pull request by rahearn**
closes #94 

Changes:

* add helpers for docker-trestle tool rather than compliance-template.
* publish control statements provided by rails-template as component-definitions
* Add github actions and circleci steps for validating and assembling SSP models.

### CODE REVIEW

File: lib/generators/rails_template18f/oscal/templates/bin/trestle.tt
Comment (pauldoomgov) [inline lib/generators/rails_template18f/oscal/templates/bin/trestle.tt:8]: suggestion (non-blocking): Handle files/dirs with spaces.

If $0 includes a space things will get sad. This fixes it at the cost of double quote mind melting.

```suggestion
oscal_location="$(dirname "$(realpath "$0")")/../doc/compliance/oscal"
```

**Review by pauldoomgov — APPROVED** [review]
I am not fluent in Ruby so you might want a second opinion, but LGTM!

File: lib/generators/rails_template18f/oscal/templates/bin/trestle.tt
Comment (rahearn) [reply lib/generators/rails_template18f/oscal/templates/bin/trestle.tt:8]: my brain can't comprehend that that double quoting works properly, but my computer seems to agree with you

**Review by zjrgov — APPROVED** [review]

**Pull request merged**
